lib/proxy/forward.c

```diff
@@ -131,6 +133,7 @@ int proxy_forward_have_authenticated(cmd_rec *cmd) {
 
   switch (proxy_method) {
     case PROXY_FORWARD_METHOD_USER_NO_PROXY_AUTH:
+    case PROXY_FORWARD_METHOD_USER_NO_PROXY_AUTH_SNI:
```

Comment:
Should this `NO_PROXY_AUTH_SNI` case check that the session is indeed using TLS, using _e.g._:
```
/* Verify that our connection is using TLS here... */
if (session.rfc2228_mech != NULL &&
    strcmp(session.rfc2228_mech, "TLS") == 0) {
```